easybuild/tools/run.py

```diff
@@ -110,21 +114,13 @@ def pad_4_spaces(msg):
             pad_4_spaces(f"working directory         ->  {self.work_dir}"),
         ]
 
-        tmpdir = tempfile.mkdtemp(prefix='shell-cmd-error-')
-        output_fp = os.path.join(tmpdir, f"{self.cmd_name}.out")
-        with open(output_fp, 'w') as fp:
-            fp.write(self.output or '')
-
-        if self.stderr is None:
-            error_info.append(pad_4_spaces(f"output (stdout + stderr)  ->  {output_fp}"))
+        if self.stderr is None and self.out_file is not None:
+            error_info.append(pad_4_spaces(f"output (stdout + stderr)  ->  {self.out_file}"))
         else:
-            stderr_fp = os.path.join(tmpdir, f"{self.cmd_name}.err")
-            with open(stderr_fp, 'w') as fp:
-                fp.write(self.stderr)
-            error_info.extend([
-                pad_4_spaces(f"output (stdout)           ->  {output_fp}"),
-                pad_4_spaces(f"error/warnings (stderr)   ->  {stderr_fp}"),
-            ])
+            if self.out_file is not None:
+                error_info.append(pad_4_spaces(f"output (stdout)           ->  {self.out_file}"))
+            if self.err_file is not None:
+                error_info.append(pad_4_spaces(f"error/warnings (stderr)   ->  {self.err_file}"))
```

Comment:
it took me a while to understand the logic behind this. Maybe it can be simplified a bit with something like:
```
        if self.out_file is not None:
            out_info_msg = "output (stdout)"
            if self.stderr is None:
                out_info_msg = "output (stdout + stderr)"
            error_info.append(pad_4_spaces(f"{out_info_msg}  ->  {self.out_file}"))

        if self.err_file is not None:
            error_info.append(pad_4_spaces(f"error/warnings (stderr)   ->  {self.err_file}"))
```

Reply:
Yeah, I guess that's clearer, fixed in 8620e64b9